spec/puppetserver/ca/sign_action_spec.rb

```diff
@@ -0,0 +1,94 @@
+require 'puppetserver/ca/sign_action'
+require 'puppetserver/ca/cli'
+require 'puppetserver/ca/logger'
+
+RSpec.describe 'Puppetserver::Ca::SignAction' do
+  let(:err)    { StringIO.new }
+  let(:out) { StringIO.new }
+  let(:logger) { Puppetserver::Ca::Logger.new(:info, out, err) }
+  let(:action) { Puppetserver::Ca::SignAction.new(logger) }
+
+  describe 'validation' do
+    before do
+      allow_any_instance_of(Puppetserver::Ca::SignAction).to receive(:run).
+          and_raise("Validation succeeded when it should have failed")
+    end
+
+    it 'does not expect an argument' do
+      exit_code = Puppetserver::Ca::Cli.run(['sign', '--all', 'foo'], out, err)
+      expect(err.string).to match(/Error:.*Unknown input.*foo/m)
+      expect(exit_code).to eq(1)
+    end
+
+    it 'does not allow --all with a valid certname call' do
+      exit_code = Puppetserver::Ca::Cli.run(['sign', '--certname', 'foo', '--all'], out, err)
+      expect(err.string).to include('--all and --certname cannot be used together')
+      expect(exit_code).to eq(1)
+    end
+
+    it 'errors without any arguments' do
+      exit_code = Puppetserver::Ca::Cli.run(['sign'], out, err)
+      expect(err.string).to include('No arguments given')
+      expect(exit_code).to eq(1)
+    end
+
+    it 'errors without a certname' do
+      exit_code = Puppetserver::Ca::Cli.run(['sign', '--certname'], out, err)
+      expect(err.string).to match(/Error:.*Missing argument to flag/m)
+      expect(exit_code).to eq(1)
+    end
+
+    it 'does not allow --certname and --all' do
+      exit_code = Puppetserver::Ca::Cli.run(['sign', '--certname', '--all'], out, err)
+      expect(err.string).to match(/Cannot use --all.*HTTP API/)
+      expect(exit_code).to eq(1)
+    end
+
+    it 'works with a single cert' do
+      results, exit_code = action.parse(['--certname', 'foo'])
```

Comment:
Is there a reason we aren't doing the failure cases by calling `parse` directly? I got kinda tripped up figuring out the purpose of the `before` block in this set of tests, and if we did them all but calling parse directly, we wouldn't need it.

Reply:
Good point. I went back and fixed all of them and removed the ``before`` block